File: api/src/models/Defect.js
const mongoose = require("mongoose");

const defectSchema = new mongoose.Schema(
  {
    raisedByTeam: {
      type: String,
      required: true,
      trim: true,
    },
    description: {
      type: String,
      required: true,
    },
    activities: {
      type: String,
      default: "",
    },
    responsible: {
      type: String,
      required: true,
      trim: true,
    },
    priority: {
      type: String,
      enum: ["Low", "Medium", "High", "Critical"],
      required: true,
      default: "Low",
    },
    dueDate: {
      type: Date,
    },
    status: {
      type: String,
      enum: ["Open", "In Progress", "Blocked", "Closed", "Reopened"],
      required: true,
      default: "Open",
    },
    nextCheck: {
      type: Date,
    },
    remark: {
      type: String,
      default: "",
    },
    comments: [
      {
        id: { type: String },
        author: {
          accountId: { type: String },
          emailAddress: { type: String },
          displayName: { type: String },
          active: { type: Boolean, default: true },
        },
        content: [
          {
            type: { type: String },
            text: { type: String },
          },
        ],
        created: { type: Date, default: Date.now },
        updated: { type: Date, default: Date.now },
      },
    ],
  },
  { timestamps: true }
);

module.exports = mongoose.model("Defect", defectSchema);

File: csharp/rebisco-bugtracker.Api/domain/defects/model/Defect.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace rebisco_bugtracker.Api.domain.defects
{
    [Table("Defects")]
    public class Defect
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        [Required]
        public string RaisedByTeam { get; set; } = string.Empty;

        [Required]
        public string Description { get; set; } = string.Empty;

        public string? Activities { get; set; }

        [Required]
        public string Responsible { get; set; } = string.Empty;

        public string? Priority { get; set; }

        public DateTime? DueDate { get; set; }

        public string? Status { get; set; }

        public DateTime? NextCheck { get; set; }

        public string? Remark { get; set; }

         public DateTime CreatedDate { get; set; }
         
          public DateTime UpdatedDate { get; set; }
    }
}